bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/TextLayout.java

```diff
@@ -3564,6 +3584,13 @@ public void setTabs (int[] tabs) {
  */
 public void setText (String text) {
 	checkLayout();
+
+	this.originalText = text;
+	String[] strings = text.split("\r");
+	text = "";
+	for (int i = 0; i < strings.length; i++)
```

Comment:
This can be done using a map reduction. Anyway, please add braces to the for loop

[reference](https://www.google.com/url?sa=t&source=web&cd=&ved=2ahUKEwjQk7ChiquHAxXHzAIHHVg0BosQFnoECA8QAQ&url=https%3A%2F%2Fwww.baeldung.com%2Fjava-stream-reduce&usg=AOvVaw0nCMK38jusEkWmmenr6Mzu&opi=89978449)

Reply:
It's this a complicated and poorly performance way of doing test.replace("\r", "")?

Reply:
@merks yes, looks like it. Your approach is the best to me!

Reply:
let me test and update

Reply:
Thanks @merks for your feedback, after updating this i found a better way of fixing the issue. So new change is updated, you may please review now.